instrumentation/active_support/test/opentelemetry/instrumentation/active_support/span_subscriber_test.rb

```diff
@@ -205,123 +205,226 @@ def finish(name, id, payload)
   end
 
   describe 'instrument' do
-    before do
-      ActiveSupport::Notifications.unsubscribe(notification_name)
+    after do
+      ActiveSupport::Notifications.notifier.all_listeners_for(notification_name).each do |listener|
+        ActiveSupport::Notifications.unsubscribe(listener)
+      end
     end
 
-    it 'does not trace an event by default' do
-      ActiveSupport::Notifications.subscribe(notification_name) do
-        # pass
+    describe 'when subscribing to events directly' do
```

Comment:
The tests in this context are the same as before, I've just added one level of indentation.